tests/StringTest.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Equip\String;
+
+use PHPUnit_Framework_TestCase as TestCase;
+
+class StringTest extends TestCase
+{
+    /**
+     * @return array
+     */
+    public function snakeStringProvider()
+    {
+        return [
+            [
+                'SnakeCase',
+                'snake_case',
+                true,
+            ],
+            [
+                'snakeCase',
+                'snake_case',
+                false,
+            ],
+            [
+                'SnakeCamelCase',
+                'snake_camel_case',
+                true,
+            ],
+            [
+                'snakeCamelCase',
+                'snake_camel_case',
+                false,
+            ],
+            [
+                'Snake',
+                'snake',
+                true,
+            ],
+            [
+                'snake',
+                'snake',
+                false,
+            ]
```

Comment:
I noticed you had some special handling for extra underscores - it might be good to see some tests of that.